src/xrpld/app/tx/detail/LoanSet.cpp

```diff
@@ -0,0 +1,632 @@
+#include <xrpld/app/tx/detail/LoanSet.h>
+//
+#include <xrpld/app/misc/LendingHelpers.h>
+
+#include <xrpl/protocol/TxFlags.h>
+
+namespace ripple {
+
+bool
+LoanSet::checkExtraFeatures(PreflightContext const& ctx)
+{
+    return checkLendingProtocolDependencies(ctx);
+}
+
+std::uint32_t
+LoanSet::getFlagsMask(PreflightContext const& ctx)
+{
+    return tfLoanSetMask;
+}
+
+NotTEC
+LoanSet::preflight(PreflightContext const& ctx)
```

Comment:
are we checking some where in preflight that counterparty isn't the same as the account?
I didn't find anything explicit check like

```
if (ctx.tx.isFieldPresent(sfCounterparty) && ctx.tx[sfCounterparty] == ctx.tx[sfAccont])
   return temINVALID; 
```